src/Driver.cpp

```diff
@@ -1,24 +1,75 @@
-#include <motors_roboteq_canopen/Driver.hpp>
 #include <motors_roboteq_canopen/Channel.hpp>
+#include <motors_roboteq_canopen/Driver.hpp>
 
 using namespace std;
 using namespace motors_roboteq_canopen;
 
 Driver::Driver(canopen_master::StateMachine& state_machine, int channel_count)
-    : DriverBase(state_machine) {
+    : DriverBase(state_machine)
+{
     for (int i = 0; i < channel_count; ++i) {
         addChannel(new Channel(*this, i));
     }
 }
 
-Channel& Driver::getChannel(int i) {
+Channel& Driver::getChannel(int i)
+{
     return static_cast<Channel&>(DriverBase::getChannel(i));
 }
 
-vector<canbus::Message> Driver::queryMotorStop() {
+vector<canbus::Message> Driver::queryMotorStop()
+{
     vector<canbus::Message> messages;
     for (size_t i = 0; i < getChannelCount(); ++i) {
         messages.push_back(getChannel(i).queryMotorStop());
     }
     return messages;
+}
+
+void Driver::writeDigitalOutput(std::uint8_t index, raw_io::Digital const& out)
+{
+    if (out.data) {
+        enableDigitalOutput(index);
+    }
+    else {
+        disableDigitalOutput(index);
+    }
+}
+
+void Driver::enableDigitalOutput(std::uint8_t index)
+{
+    set<ActivateDigitalOutput>(index);
```

Comment:
This returns a canbus::Message that the class client has to write on the bus (as SDO)